inspector/lib.py

```diff
@@ -133,6 +134,9 @@ def should_start(task: Task, data_dir: str | os.PathLike, srv) -> bool:
         mem_gib = srv.memory_amount / 1024
         logging.info(f"Skipping task {task.name} because it requires {task.minimum_memory} GiB RAM, but this machine has only {mem_gib:.03}")
         return False
+    if task.servers_only:
+        if not any((srv.vendor_id, srv.api_reference) == so for so in task.servers_only):
+            return False
```

Comment:
```suggestion
    if task.servers_only and (srv.vendor_id, srv.api_reference) not in task.servers_only:
        return False
```
I might be overlooking something, but isn't this enough?
Also, this check runs during `inspector.start`, you'll need to add the same to `inspector.inspect`, which calls `should_run`.

Reply:
I was tempted to do that latter by looking up vendor/instance from the path :smile: 
but passed down the 2 variables after all .. i hope it looks ok